magento/pwa-studio
# [PWA-1658] Move graphql-cli-validate-magento-pwa-queries to @magento scope

## Description

As part of https://jira.corp.magento.com/browse/PWA-1602 we realized that all our packages except graphql-cli-validate-magento-pwa-queries are under the @magento custom scope.

Using custom scopes safe guards us against supply chain attacks. So as part of this PR, lets move graphql-cli-validate-magento-pwa-queries under magento scope and deprecate the old package.

Acceptance Criteria

Publish the package to the new namespace / scope
Submit a PR pointing any existing code in PWA Studio to the new package, to be released in the next PWA Studio release
Communicate the upcoming deprecation to stakeholders, including community
(Ideally after next release) post a deprecation notice / deprecate the existing package that is not in the magento scope

## Proposed Solution

`graphql-cli@3` requires plugin names begin with `graphql-cli-`, so re-naming was not going to be possible. It seemed achievable that upgrading to `graphql-cli@4` would mean we could leverage their new plugin framework, but from my discovery the naming requirement now meant you had to be in the `@graphql-cli` namespace. Before digging too much into that solution, I instead focused on what it would take to get `graphql validate` working in the project.

The solution proposed has some flaws, and is a tiny bit less feature rich, but accomplishes the same task. It also gives us the opportunity to start leveraging some cooler features of `graphql-cli`, like type generation, or change how we approach operation creation.

## Related Issue





* [[PWA-1658](https://jira.corp.magento.com/browse/PWA-1658)] Move graphql-cli-validate-magento-pwa-queries to @magento scope

## Related PRs
* https://github.com/magento-commerce/pwa-studio-cicd/pull/73

## Acceptance



### Verification Stakeholders

### Specification



### Verification Steps


1. `yarn validate-queries` - verify no errors returned (expected to see lots of deprecation warnings)
2. `yarn lint` - verify no errors

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.
